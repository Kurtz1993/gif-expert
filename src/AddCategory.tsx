import { ChangeEvent, FormEvent, useState } from 'react';

export default function AddCategory() {
  const [inputValue, setInputValue] = useState('Hola');

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    console.log(e);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
      <button type="submit">Add Category</button>
    </form>
  );
}
