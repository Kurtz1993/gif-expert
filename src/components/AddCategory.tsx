import { ChangeEvent, Dispatch, FormEvent, SetStateAction, useState } from 'react';

interface AddCategoryProps {
  setCategories: Dispatch<SetStateAction<string[]>>;
}

export default function AddCategory({ setCategories }: AddCategoryProps) {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
      <button type="submit">Add Category</button>
    </form>
  );
}
