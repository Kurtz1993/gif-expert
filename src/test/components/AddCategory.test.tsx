import { mount, ReactWrapper } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('AddCategoryComponent', () => {
  let wrapper: ReactWrapper;

  const setCategories = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = mount(<AddCategory setCategories={setCategories} />);
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the textbox', () => {
    const input = wrapper.find('input');
    const expectedText = 'hello';

    input.simulate('change', { target: { value: expectedText } });
  });

  test('should not POST the information onSubmit', () => {
    const form = wrapper.find('form');

    form.simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call "setCategories" and clean input onSubmit', () => {
    const input = wrapper.find('input');
    const form = wrapper.find('form');
    const expectedText = 'hello';

    input.simulate('change', { target: { value: expectedText } });
    form.simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
