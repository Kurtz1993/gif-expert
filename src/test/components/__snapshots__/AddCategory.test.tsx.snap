// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategoryComponent should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add Category
  </button>
</form>
`;
